test/geocel/GeoTests.cc

```diff
@@ -366,6 +366,62 @@ void FourLevelsGeoTest::test_trace() const
     }
 }
 
+//---------------------------------------------------------------------------//
+// LAR SPHERE
+//---------------------------------------------------------------------------//
+void LarSphereGeoTest::test_trace() const
+{
+    bool const is_orange = test_->geometry_type() == "ORANGE";
+    {
+        SCOPED_TRACE("+y");
+        auto result = test_->track({0, -120, 0}, {0, 1, 0});
+
+        GenericGeoTrackingResult ref;
+        ref.volumes = {
+            "world",
+            "detshell_bot",
+            "sphere",
+            "detshell_top",
+            "world",
+        };
+        ref.volume_instances = {
+            "world_PV",
+            "detshell_bot_PV",
+            "sphere_PV",
+            "detshell_top_PV",
+            "world_PV",
+        };
+        ref.distances = {10, 10, 200, 10, 890};
+        ref.halfway_safeties = {5, 5, 100, 5, 445};
+        ref.bumps = {};
+        if (is_orange)
+        {
+            // TODO: at this exact point it ignores the spherical distance
+            ref.halfway_safeties[2] = result.halfway_safeties[2];
```

Comment:
Why is this?

Reply:
The ORANGE safety calculation currently calculates the gradient of the quadric and finds the distance to boundary along that direction. Unfortunately at the exact center of the sphere, the gradient is zero so there's no intersection.